osu.Game/Storyboards/IStoryboardElementHasDuration.cs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Storyboards
+{
+    /// <summary>
+    /// A StoryboardElement that ends at a different time than its start time.
+    /// </summary>
+    public interface IStoryboardElementHasDuration
```

Comment:
I think it would be far better if this becomes a "with duration" interface instead:
```csharp
public interface IStoryboardElementWithDuration : IStoryboardElement
{
     double EndTime { get; }
}
```

Reply:
Also now that I think about it, it's a bit weird that this interface has an `EndTime` member but no `Duration`. Especially that it's a "storyboard element with duration".

Reply:
Should I move `EndTime` back to `IStoryboardElement`?

Reply:
no it's to add another member `Duration` just like in `IHasDuration`.